### PR TITLE
[MOD-2311] Android: add Map styling via json

### PR DESCRIPTION
**Pull request by m1ga**
Jira: https://jira.appcelerator.org/browse/MOD-2311

Add setMapStyle to style the map via json from https://mapstyle.withgoogle.com/

demo

```css
"#mapview": {
    style:'[{"elementType":"geometry","stylers":[{"color":"#242f3e"}]},{"elementType":"labels.icon","stylers":[{"visibility":"off"}]},{"elementType":"labels.text.fill","stylers":[{"color":"#746855"}]},{"elementType":"labels.text.stroke","stylers":[{"color":"#242f3e"}]},{"featureType":"administrative","elementType":"geometry","stylers":[{"color":"#757575"}]},{"featureType":"administrative.country","elementType":"labels.text.fill","stylers":[{"color":"#9e9e9e"}]},{"featureType":"administrative.land_parcel","stylers":[{"visibility":"off"}]},{"featureType":"administrative.locality","elementType":"labels.text.fill","stylers":[{"color":"#d59563"}]},{"featureType":"poi","elementType":"labels.text","stylers":[{"visibility":"off"}]},{"featureType":"poi","elementType":"labels.text.fill","stylers":[{"color":"#d59563"}]},{"featureType":"poi.business","stylers":[{"visibility":"off"}]},{"featureType":"poi.park","elementType":"geometry","stylers":[{"color":"#263c3f"}]},{"featureType":"poi.park","elementType":"labels.text.fill","stylers":[{"color":"#6b9a76"}]},{"featureType":"poi.park","elementType":"labels.text.stroke","stylers":[{"color":"#1b1b1b"}]},{"featureType":"road","stylers":[{"visibility":"off"}]},{"featureType":"road","elementType":"geometry","stylers":[{"color":"#38414e"}]},{"featureType":"road","elementType":"geometry.fill","stylers":[{"color":"#2c2c2c"}]},{"featureType":"road","elementType":"geometry.stroke","stylers":[{"color":"#212a37"}]},{"featureType":"road","elementType":"labels.icon","stylers":[{"visibility":"off"}]},{"featureType":"road","elementType":"labels.text.fill","stylers":[{"color":"#9ca5b3"}]},{"featureType":"road.arterial","elementType":"geometry","stylers":[{"color":"#373737"}]},{"featureType":"road.highway","elementType":"geometry","stylers":[{"color":"#746855"}]},{"featureType":"road.highway","elementType":"geometry.stroke","stylers":[{"color":"#1f2835"}]},{"featureType":"road.highway","elementType":"labels.text.fill","stylers":[{"color":"#f3d19c"}]},{"featureType":"road.highway.controlled_access","elementType":"geometry","stylers":[{"color":"#4e4e4e"}]},{"featureType":"road.local","elementType":"labels.text.fill","stylers":[{"color":"#616161"}]},{"featureType":"transit","stylers":[{"visibility":"off"}]},{"featureType":"transit","elementType":"geometry","stylers":[{"color":"#2f3948"}]},{"featureType":"transit","elementType":"labels.text.fill","stylers":[{"color":"#757575"}]},{"featureType":"transit.station","elementType":"labels.text.fill","stylers":[{"color":"#d59563"}]},{"featureType":"water","elementType":"geometry","stylers":[{"color":"#17263c"}]},{"featureType":"water","elementType":"labels.text.fill","stylers":[{"color":"#515c6d"}]},{"featureType":"water","elementType":"labels.text.stroke","stylers":[{"color":"#17263c"}]}]'
}

```

### CODE REVIEW

File: android/src/ti/map/TiUIMapView.java [hunk 1]
```diff
@@ -172,6 +177,17 @@ public void onMapReady(GoogleMap gMap) {
...
+		if (styleString != null){
+			try {
+	            boolean success = map.setMapStyle(new MapStyleOptions(styleString));
+	            if (!success) {
+	                Log.e("MapsActivityRaw", "Style parsing failed.");
+	            }
+	        } catch (Resources.NotFoundException e) {
+	            Log.e("MapsActivityRaw", "Can't find style.", e);
+	        }
+		}
```
Comment (hansemannn) [inline android/src/ti/map/TiUIMapView.java]: Indentation of the try-catch

File: apidoc/View.yml [hunk 2]
```diff
@@ -684,6 +684,15 @@ properties:
     since: "5.2.0"
     platforms: [iphone, ipad]
 
+  - name: style
+    summary: JSON String to style the Map.
+    description: |
+        This property will change the look (colors, roads, labels) of the map. A valid JSON can be created at (Google Maps Styling Wizard)[https://mapstyle.withgoogle.com/]
```
Comment (hansemannn) [inline apidoc/View.yml]: break the line after "can be created"

Comment (hansemannn) [inline apidoc/View.yml]: And please bump the version

**Comment by m1ga** [comment]
Added getter/setter and updated the example on JIRA.

File: apidoc/View.yml [hunk 3]
```diff
@@ -684,6 +684,16 @@ properties:
     since: "5.2.0"
     platforms: [iphone, ipad]
 
+  - name: style
+    summary: JSON String to style the Map.
+    description: |
+        This property will change the look (colors, roads, labels) of the map. A valid JSON can be created
+        at (Google Maps Styling Wizard)[https://mapstyle.withgoogle.com/]
```
Comment (hansemannn) [inline apidoc/View.yml]: And please update the changelog in `android/documentation/changelog.md`

**Comment by hansemannn** [comment]
LGTM, handing over to @garymathews to finish review.

**Comment by m1ga** [comment]
ready to test!

**Comment by garymathews** [comment]
CR: **PASS**
FT: **PASS**

👍

**Pull request merged**